pkg/chains/storage/docdb/docdb_test.go

```diff
@@ -91,21 +91,21 @@ func TestBackend_StorePayload(t *testing.T) {
 			}
 
 			// Check the signature.
-			sig, err := b.RetrieveSignature(opts)
+			signatures, err := b.RetrieveSignatures(opts)
 			if err != nil {
 				t.Fatal(err)
 			}
-			if string(sig) != tt.args.signature {
-				t.Errorf("wrong signature, expected %s, got %s", tt.args.signature, string(sig))
+			if signatures[obj.Name][0] != tt.args.signature {
```

Comment:
let's add a check to make sure there's exactly one signature here as well,

```go
if len(signatures[obj.Name]) != 1 {
  t.Fatalf("unexpected number of signatures: %d", len(signatures[obj.Name]))
}
```

Reply:
I can add that check to the tests for sure, but in this case shouldn't we rewrite the `RetrieveSignatures` function with the check instead?

Something like:

```go
func (b *Backend) RetrieveSignatures(opts config.StorageOpts) (map[string][]string, error) {
	// Retrieve the document.
	documents, err := b.retrieveDocuments(opts)
	if err != nil {
		return nil, err
	}

        if len(documents) != 1 {
            return nil, fmt.Errorf("unexpected number of signatures: %d", len(documents))
        }

	m := make(map[string][]string)
	// Extract and decode the signature.
	sig, err := base64.StdEncoding.DecodeString(d.Signature)
	if err != nil {
		return nil, err
	}
	m[d.Name] = []string{string(sig)}
	return m, nil
}
```

Reply:
i think it's ok to leave the check in the test for now!

Reply:
Copy that 👍